src/GraphQL/Instrumentation/FieldResolverBuilderExtensions.cs

```diff
@@ -8,42 +8,72 @@ namespace GraphQL.Instrumentation
 {
     public static class FieldResolverBuilderExtensions
     {
-        private const string InvokeMethodName = "Resolve";
+        private const string INVOKE_METHOD_NAME = "Resolve";
 
-        public static IFieldMiddlewareBuilder Use<T>(this IFieldMiddlewareBuilder builder) where T : new() => Use(builder, typeof(T));
+        public static IFieldMiddlewareBuilder Use(this IFieldMiddlewareBuilder builder, IFieldMiddleware middleware)
+            => builder.Use(next => context => middleware.Resolve(context, next));
```

Comment:
I like the added support for creating middleware that is "statically linked", and does not rely on DI at runtime.  I did notice that the default DI provider included with GraphQL performs identical to previously (as it pertains to middleware), with Activator.CreateInstance